app/email.py

```diff
@@ -1,21 +1,16 @@
-from threading import Thread
-
-from flask import current_app, render_template
+import os
+from flask import render_template
 from flask.ext.mail import Message
 from . import mail
+from app import create_app
 
 
-def send_async_email(app, msg):
+def send_async_email(recipient, subject, template, **kwargs):
+    app = create_app(os.getenv('FLASK_CONFIG') or 'default')
     with app.app_context():
+        msg = Message(app.config['EMAIL_SUBJECT_PREFIX'] + ' ' + subject,
+                      sender=app.config['EMAIL_SENDER'],
+                      recipients=[recipient])
+        msg.body = render_template(template + '.txt', **kwargs)
+        msg.html = render_template(template + '.html', **kwargs)
         mail.send(msg)
```

Comment:
That's a good point. This function is meant to be used off of the main thread (hence why I create a new flask app) but in theory it could be used synchronously.